E2E: Refactor waitTillPresentAndDisplayed helper & review its usage

WunderBart opened the pull request:
#### Changes proposed in this Pull Request

- Refactored `waitTillPresentAndDisplayed` helper
  - Renamed to `waitUntilLocatedAndVisible` for better alignment with native helpers ([`elemenLocated`](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/until.html#elementLocated) & [`elementIsVisible`](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/lib/until.html#elementIsVisible))
  - Fixed issue where the `isDisplayed` promise result [was overridden by `true`](https://github.com/Automattic/wp-calypso/pull/51506/files#diff-4b1fd2b061afbd8739cfc7b385f436817a04cb31125fadcda287711048263968L95), making the helper always be positive about the element visibility, regardless of the factual state. This discovery caused a bit larger refactor as there were quite a few elements that were expected to be visible but weren't in reality. I had to review each one of them and determine if the visibility is broken and provide a fix. Thankfully, it turned out that the app is fine and only E2Es needed some updating.
  - Removed the helper from where it was used unnecessarily (i.e. when followed by the `clickWhenClickable` helper, which determines the interactability of the same element)
  - Updated docs

#### Testing instructions

All tests should pass. (_Edit: That one failing flaky desktop spec is unrelated. I'm going to address it in another PR._)

WunderBart committed:
Refactor waitTillPresentAndDisplayed helper & review it's usage

WunderBart committed:
Add & update docs

WunderBart committed:
Remove obsolete waiter

WunderBart committed:
Fix specs that want invisible elements to be visible

WunderBart committed:
Do not trigger `viewPublishedPostOrPage` click programmatically

WunderBart committed:
Wait for content switch when previewing post

WunderBart committed:
Remove obsolete check

WunderBart committed:
Merge branch 'trunk' into refactor/e2e-tweaks-part-021

WunderBart committed:
Don't wait for closed (invisible) sidebar
...just wait for the parent element instead. There's a post-init that checks ensures ot to be open anyway.

WunderBart committed:
Do not ensure that sidebar is open - it's ensured in post-init

WunderBart committed:
Fix username getter that was always silently failing

WunderBart committed:
Remove accidental describe.only

WunderBart committed:
Merge branch 'trunk' into refactor/e2e-tweaks-part-021

WunderBart committed:
Merge branch 'trunk' into refactor/e2e-tweaks-part-021

WunderBart committed:
Fix username retrieval from account settings page

WunderBart committed:
Fix profile page component viewport-specific interactions

WunderBart committed:
Update purchases page expected element

WunderBart commented:
FYI, that one failing flaky desktop spec is unrelated. I'm going to address it in another PR.

bsessions85 reviewed (approved):
looks good! Now I'll just have to remember to use the new one! Thanks

noahtallen commented on test/e2e/docs/style-guide.md:
```diff
@@ -25,21 +25,23 @@ We don't chain function calls together and avoid using `.then` calls.
+	const modalButtonLocator = By.css( 'button.open-modal' );
+	const modalLocator = By.css( '.modal' );
+	await driverHelper.clickWhenClickable( this.driver, modalButtonLocator ).then( async function () {
+		await driverHelper.waitUntilLocatedAndVisible( modalLocator );
```
IMO, the previous example might be a bit better because it's relying on promises much more -- I think people would be much less likely to implement async/await inside a promise callback like this just because part of the point of async/await is to avoid `.then`.) In other words the previous example is like "don't use promises directly", and this one is like "don't do async/await poorly". What do you think?

WunderBart replied:
Good point, I agree! 🙌  Will remove the async from the callback 👍

WunderBart replied:
Updated in https://github.com/Automattic/wp-calypso/pull/51506/commits/9273124232190539f4e775a9c06e9668472710b6

noahtallen reviewed (approved):
awesome!

noahtallen commented:
This might need a rebase, I think the e2e test suite didn't run

WunderBart committed:
Merge branch 'trunk' into refactor/e2e-tweaks-part-021

WunderBart committed:
Update e2e style guide example
As per https://github.com/Automattic/wp-calypso/pull/51506/files/19eef2275b4a0f8fa075acf340d8ed6d42c3aea6#r610952596
